src/frontend/src/app/myprofile/myprofile.component.ts

```diff
@@ -10,24 +12,46 @@ import { AuthService } from '../services/auth/auth.service';
 })
 export class MyProfileComponent {
   player: PlayerData;
-
-  constructor(private router: Router, private authService: AuthService) {
+  userStats: Stats[] | undefined;
+  constructor(
+    private router: Router,
+    private authService: AuthService,
+    private profileService: ProfileService,
+    private problemService: ProblemsService
+  ) {
     this.player = {
       id: 'a',
       name: '',
       picture: '',
       finishedCurrentRound: true,
-      wins: 10,
-      losses: 20,
-      problemSolved: 30,
-      easy: 10,
-      medium: 10,
-      hard: 10,
+      wins: 0,
+      losses: 0,
+      problemSolved: 0,
     } as PlayerData;
     const playerStats = authService.getPlayerData();
     this.player.picture = playerStats.picture;
     this.player.name = playerStats.name;
   }
+  ngOnInit(): void {
+    this.profileService.getUserStats().subscribe((res) => {
+      console.log(res);
+      this.userStats = res;
+      let won = 0;
+      let problemSolved = 0;
+      this.userStats.forEach((element) => {
+        if (element.win) {
+          won++;
+        }
+        problemSolved += element.problems.length;
+      });
+      const tempPlayerStats = this.player;
+      tempPlayerStats.wins = won;
+      tempPlayerStats.losses = this.userStats.length - won;
+      tempPlayerStats.problemSolved = problemSolved;
+      this.player = tempPlayerStats;
```

Comment:
Directly update the player, no need for tempPlayerStats

```suggestion
      this.player.wins = won;
      this.player.losses = this.userStats.length - won;
      this.player.problemSolved = problemSolved;
```

Reply:
Temp is needed